backend/main/chapters/c09_combining_booleans.py

```diff
@@ -0,0 +1,638 @@
+
+        def program(self):
+            def is_friend(name):
+                return name == "Alice" or "Bob"
+
+            assert_equal(is_friend("Alice"), True)
+            assert_equal(is_friend("Bob"), True)
+            assert_equal(is_friend("Charlie"), False)
+
+    class InspectWithBirdseye(ACommonMistake):
+        text = """
+The second and third tests fail! Our function seems to be doing the wrong thing:
+it returns `Bob` (a string, not a boolean!) when `name` is `Charlie`. What is going on?
+Try inspecting the code with Birdseye. Inspect the `return` statements of each `is_friend` calls carefully.
+(Use the blue arrow buttons)
+        """
+
+        def check(self):
+            return super().check() and self.code_source == "birdseye"
+
+    class AnExercise(ExerciseStep):
+        """
+When we inspect it with Birdseye, we can see that:
+
+    name == "Alice" or "Bob"
+
+is not translated into
+
+    name == ("Alice" or "Bob")
+
+the way we think in English, but rather:
+
+    (name == "Alice") or ("Bob")
+
+What then happens is:
+
+- `name == "Alice"` evaluates to `False`,
+- `name == "Alice" or "Bob"` evaluates to `"Bob"`, and
+-  `return "Bob"` returns the string "Bob", instead of a boolean value, which is what we originally intended.
+
+Perhaps you feel like this:
+
+[![I now have additional questions](https://i.imgur.com/jN57tGt.png)](https://imgur.com/a/icKzI)
+
+The only thing you really need to know is this: Until you know what you're doing, always
+make sure you put booleans on both sides of `or`, because it's a boolean operator.
+`name == "Alice" or "Bob"` breaks that rule.
+
+If you're curious, the answers are below, but you can skip them if you want.
+Make sure to read the Exercise that comes afterwards!
+
+----
+
+> Why does `(name == "Alice") or ("Bob")` equal `"Bob"`? Why does it equal anything? `"Bob"` isn't even a boolean!
+
+The definition "`A or B` is `True` if either `A` or `B` is `True`" was a simplification. It's the easiest
+way to think about `or` most of the time, especially for writing `if` statements.
+The real definition is that if `A` is true then `A or B` is just `A` (in fact `B` is not even evaluated),
+otherwise it's `B`.
+You can see for yourself that if `A` and `B` are booleans then the two definitions are equivalent.
+In this example `A` is `name == "Alice"` which is `False`, so `A or B` is `B` which is `"Bob"`.
+
+> Why does `if "Bob"` run the body? Again, `"Bob"` isn't a boolean!
+
+Python actually lets you treat anything like a boolean. Most things are equivalent to `True` in this context,
+including all strings (such as `"Bob"`) except the empty string which is 'falsy'.
+
+> Is there a better way to write the condition without repeating `name ==` each time?
+
+Yes! In [Functions and Methods for Lists](/course/?page=FunctionsAndMethodsForLists) we mentioned the `in`
+operator, which you can use with a list like this:
+
+    if name in ["Alice", "Bob", "Charlie"]:
```

Comment:
```suggestion
    return name in ["Alice", "Bob", "Charlie"]:
```

Reply:
Right, this is left over from the previous incarnation, must have missed it.